src/lib/Liblog/pbs_log.c

```diff
@@ -334,6 +374,265 @@ log_init(void)
 #endif
 }
 
+/** 
+ * @brief
+ *      Add general debugging information in log
+ *
+ * @par Side Effects:
+ * 	None
+ * 
+ * @par MT-safe: Yes
+ *
+ */
+void
+log_add_debug_info()
+{
+	char tbuf[LOG_BUF_SIZE];
+	char temp[LOG_BUF_SIZE];
+	/* Temporary buffers to create log entry string for leaf name and mom node name */
+
+	char host[PBS_MAXHOSTNAME+1];
+
+	if (!gethostname(host, (sizeof(host) - 1))) {
+		strcpy(temp, host);
+		if (!get_fullhostname(host, host, (sizeof(host) - 1)))
+			strcpy(temp, host);  /* if full hostname is available, then overwrite */
+	}
+	else
+		strcpy(temp, "N/A");
+	snprintf(tbuf, LOG_BUF_SIZE, "hostname=%s;", temp);
+
+
+	/* To add leaf node name, if set */
+	strcpy(temp, "pbs_leaf_name=");
+	if(pbs_conf.pbs_leaf_name){ 
+		strcat(temp, pbs_conf.pbs_leaf_name);
+		strcat(temp, ";");
+	}
+	else
+		strcat(temp, "N/A;");
+	strncat(tbuf, temp, LOG_BUF_SIZE);
+
+	
+	/* To add mom node name, if set */
+	strcpy(temp, "pbs_mom_node_name=");
+	if(pbs_conf.pbs_mom_node_name)
+		strcat(temp, pbs_conf.pbs_mom_node_name);
+	else
+		strcat(temp, "N/A");
+	strncat(tbuf, temp, LOG_BUF_SIZE);
+	log_record(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_INFO, msg_daemonname, tbuf);
+	
+	return;	
+
+}
+
+/**
+ * @brief
+ *      Add interface information to log
+ *
+ * @par Side Effects:
+ *      None
+ *
+ * @par MT-safe: Yes
+ *
+ */
+
+int
+log_add_if_info()
+{
+	char tbuf[LOG_BUF_SIZE];
+	char inet_family[10];
+#if defined(linux)
+
+	int i, ret;
+	char **hostnames;
+	struct ifaddrs *ifp, *listp;
+
+	ret = getifaddrs(&ifp);
+	if ((ret != 0) || (ifp == NULL)) {
+		fprintf(stderr, "ERROR: Failed to obtain interface names.\n");
+		return 1;
+	}
+	for (listp = ifp; listp; listp = listp->ifa_next) {
+		hostnames = get_if_hostnames(listp->ifa_addr);
+
+		get_sa_family(listp->ifa_addr, inet_family);
+		snprintf(tbuf, LOG_BUF_SIZE, "%s interface ", inet_family);
+
+		if (!hostnames)
+			continue;
+		strcat(tbuf, listp->ifa_name);
+		strcat(tbuf, ": ");
+		for (i = 0; hostnames[i]; i++){
+			strcat(tbuf, hostnames[i]);
+			strcat(tbuf, " ");
+		}
+		log_record(PBSEVENT_SYSTEM, PBS_EVENTCLASS_SERVER, LOG_INFO, msg_daemonname, tbuf);
+		free_if_hostnames(hostnames);
+	}
+	freeifaddrs(ifp);
+
+#elif defined(WIN32)
+
+	int i;
+	char **hostnames;
+	PIP_ADAPTER_ADDRESSES addrlistp, addrp;
+	PIP_ADAPTER_UNICAST_ADDRESS ucp;
+	DWORD size = 8192;
+	DWORD ret;
+	WSADATA wsadata;
+
+	addrlistp = (IP_ADAPTER_ADDRESSES *)malloc(size);
+	if (!addrlistp) {
+		fprintf(stderr, "ERROR: Out of memory.\n");
+		return 1;
+	}
+	ret = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, addrlistp, &size);
+	if (ret == ERROR_BUFFER_OVERFLOW) {
+		addrlistp = realloc(addrlistp, size);
+		if (!addrlistp) {
+			fprintf(stderr, "ERROR: Out of memory.\n");
+			return 1;
+		}
+	}
+	ret = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, NULL, addrlistp, &size);
```

Comment:
This line should be within the above if statement. No need to call it again if we did not encounter an ERROR_BUFFER_OVERFLOW.